premise-portfolio.php

```diff
@@ -152,37 +152,56 @@ public function do_hooks() {
 		// add_action( 'init', array( PWPP_Options_page::get_instance(), 'init' ) );
 
 		// Initiate and register our custom post type
-		add_action( 'init', array( PWPP_Portfolio_CPT::get_instance(), 'init' ) );
+		$portfolio_cpt = PWPP_Portfolio_CPT::get_instance();
+
+		$portfolio_cpt->init();
 
 		// register our shortcode
 		add_shortcode( 'pwpp_portfolio', array( PWPP_Shortcode::get_instance(), 'init' ) );
 
 		add_filter( 'template_include', array( PWPP_Portfolio_CPT::get_instance(), 'portfolio_page_template' ), 99 );
 
 		add_filter( 'excerpt_length', array( PWPP_Portfolio_CPT::get_instance(), 'portfolio_excerpt_trim' ) );
+
+		// Add rewrite flush rules on init with a higher priority than 10.
+		add_action( 'init', array( $this, 'maybe_flush_rules' ), 11 );
+	}
+
+
+
+
+
+	/**
+	 * Flush rewrite rules if our plugin was just activated.
+	 */
+	public function maybe_flush_rules() {
+		// If this option exists we just activated the plugin,
+		// And if Premise-WP plugin activated too, flush rewrite rules.
+		if ( get_option( '_pwpp_activation_happened' )
+			&& class_exists( 'PremiseCPT' ) ) {
```

Comment:
@mvallejo3 checking for the PremiseCPT class here will flush rewrite rules ONLY if the PremiseWP has been activated.